Add allow-list functionality to securitychecker_enlightn

peterjaap opened the pull request:
| Q             | A
| ------------- | ---
| Branch        | v2.x
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | yes

peterjaap committed:
Add allow-list functionality to securitychecker_enlightn

peterjaap committed:
Update enlightn.md

veewee commented on src/Task/SecurityCheckerEnlightn.php:
```diff
@@ -50,6 +52,11 @@ public function run(ContextInterface $context): TaskResultInterface
         $arguments = $this->processBuilder->createArgumentsForCommand('security-checker');
         $arguments->add('security:check');
         $arguments->addOptionalArgument('%s', $config['lockfile']);
+        if (!empty($config['allow_list'])) {
```
You should use `$arguments->addArgumentArrayWithSeparatedValue()` for this specific purpose (instead of the empty + foreach + broken optional argument statement)

veewee reviewed (changes requested):
Looks like a nice addition, thanks!
Can you make sure to add a test-case for this new allow_list argument? Otherwise we won't be able to merge this.

peterjaap committed:
Merge branch 'phpro:v2.x' into patch-2

peterjaap committed:
Use addArgumentArrayWithSeparatedValue()

peterjaap committed:
Added test case for usage of --allow-list

peterjaap commented:
@veewee I've added a test case and used `addArgumentArrayWithSeparatedValue()`.

veewee commented on src/Task/SecurityCheckerEnlightn.php:
```diff
@@ -50,6 +52,7 @@ public function run(ContextInterface $context): TaskResultInterface
         $arguments = $this->processBuilder->createArgumentsForCommand('security-checker');
         $arguments->add('security:check');
         $arguments->addOptionalArgument('%s', $config['lockfile']);
+        $arguments->addArgumentArrayWithSeparatedValue('--allow-list', $config['allow_list'] ?? []);
```
allow_list is not nullable since it has a default of []

veewee commented on test/Unit/Task/SecurityCheckerEnlightnTest.php:
```diff
@@ -107,5 +107,16 @@ public function provideExternalTaskRuns(): iterable
+            [
+                'security:check',
+                './composer.lock',
+                '--allow-list=allow_advisory_1,allow_advisory_2'
```
This is not how addArgumentArrayWithSeparatedValue works.
If you want this output, you could use something like:

```php
addOptionalCommaSeparatedArgument('--allow-list=%s', $config['allow_list']);
```

veewee commented on doc/tasks/securitychecker/enlightn.md:
```diff
@@ -19,6 +19,10 @@ grumphp:
         securitychecker_enlightn:
             lockfile: ./composer.lock
             run_always: false
+            allow_list:
```
This configuraiton option should be explained in more details underneath.

veewee commented on src/Task/SecurityCheckerEnlightn.php:
```diff
@@ -24,10 +24,12 @@ public static function getConfigurableOptions(): ConfigOptionsResolver
         $resolver->setDefaults([
             'lockfile' => './composer.lock',
             'run_always' => false,
+            'allow_list' => []
```
The method `provideConfigurableOptions` in the test should be altered to cover this newly added default.

veewee reviewed (changes requested):
Thanks. It's curently not working as expected.
I've added some additional pointers in code.

leonhelmus commented:
@peterjaap nice addition. This is something i could use as well. Thanks for adding it. When will you have time to look at the changes needed?

peterjaap committed:
Add --allow-list option to SecurityCheckerEnlightn task

- Document new option in enlightn.md with example usage

- Fix argument formatting to use comma-separated values for --allow-list

- Update unit test to include the new option in default configuration

This change allows users to specify CVEs that should be ignored during security checks when they've been assessed and determined not to pose a risk.

peterjaap commented:
@veewee I processed your remarks

veewee commented:
Thanks
